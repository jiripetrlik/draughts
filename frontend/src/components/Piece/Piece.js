import React from "react"
import "./Piece.css"
import whitepiece from "./white-piece.svg"
import blackpiece from "./black-piece.svg"
import whitequeen from "./white-queen.svg"
import blackqueen from "./black-queen.svg"

class Piece extends React.Component {
    render() {
        let image
        if (this.props.type === "white-piece") {
            image = <img src={whitepiece} alt=""/>
        } else if (this.props.type === "black-piece") {
            image = <img src={blackpiece} alt=""/>
        } else if (this.props.type === "white-queen") {
            image = <img src={whitequeen} alt=""/>
        } else if (this.props.type === "black-queen") {
            image = <img src={blackqueen} alt=""/>
        }

        return image
    }
}

export default Piece;
