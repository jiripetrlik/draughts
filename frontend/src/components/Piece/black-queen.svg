<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!-- Created with Inkscape (http://www.inkscape.org/) -->

<svg
   xmlns:dc="http://purl.org/dc/elements/1.1/"
   xmlns:cc="http://creativecommons.org/ns#"
   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
   xmlns:svg="http://www.w3.org/2000/svg"
   xmlns="http://www.w3.org/2000/svg"
   xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
   xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
   width="50"
   height="50"
   viewBox="0 0 13.229166 13.229167"
   version="1.1"
   id="svg8"
   inkscape:version="0.92.4 (5da689c313, 2019-01-14)"
   sodipodi:docname="black-queen.svg">
  <defs
     id="defs2" />
  <sodipodi:namedview
     id="base"
     pagecolor="#ffffff"
     bordercolor="#666666"
     borderopacity="1.0"
     inkscape:pageopacity="0"
     inkscape:pageshadow="2"
     inkscape:zoom="11.2"
     inkscape:cx="6.810529"
     inkscape:cy="19.128015"
     inkscape:document-units="mm"
     inkscape:current-layer="layer1"
     showgrid="false"
     units="px"
     inkscape:window-width="1920"
     inkscape:window-height="997"
     inkscape:window-x="0"
     inkscape:window-y="27"
     inkscape:window-maximized="1" />
  <metadata
     id="metadata5">
    <rdf:RDF>
      <cc:Work
         rdf:about="">
        <dc:format>image/svg+xml</dc:format>
        <dc:type
           rdf:resource="http://purl.org/dc/dcmitype/StillImage" />
        <dc:title></dc:title>
      </cc:Work>
    </rdf:RDF>
  </metadata>
  <g
     inkscape:label="Vrstva 1"
     inkscape:groupmode="layer"
     id="layer1"
     transform="translate(0,-283.77082)">
    <circle
       id="path3734"
       cx="6.6145835"
       cy="290.38541"
       style="stroke-width:0.27214286"
       r="5.953125" />
    <circle
       style="fill:#ffffff;stroke:#ffffff;stroke-width:0.48870537;stroke-miterlimit:4;stroke-dasharray:none"
       id="path1410"
       cx="6.6145835"
       cy="290.38541"
       r="2.1368973" />
  </g>
</svg>
