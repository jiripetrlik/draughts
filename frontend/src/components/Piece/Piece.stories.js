import React from 'react'
import Piece from './Piece'

export default {
  component: Piece,
  title: 'Piece',
};

const Template = args => <Piece {...args} />;

export const WhitePiece = Template.bind({});
WhitePiece.args = {
  type: "white-piece"
}

export const BlackPiece = Template.bind({});
BlackPiece.args = {
  type: "black-piece"
}

export const WhiteQueen = Template.bind({});
WhiteQueen.args = {
  type: "white-queen"
}

export const BlackQueen = Template.bind({});
BlackQueen.args = {
  type: "black-queen"
}

export const Move = Template.bind({});
Move.args = {
  type: "move"
}
