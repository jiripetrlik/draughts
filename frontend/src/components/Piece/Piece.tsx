import React from "react"
import "./Piece.css"
import whitepiece from "./white-piece.svg"
import blackpiece from "./black-piece.svg"
import whitequeen from "./white-queen.svg"
import blackqueen from "./black-queen.svg"
import move from "./possible-move.svg"

class Piece extends React.Component<Readonly<any>, Readonly<any>> {
    render() {
        let image
        if (this.props.type === "white-piece") {
            image = <img src={whitepiece} alt="" className="piece"/>
        } else if (this.props.type === "black-piece") {
            image = <img src={blackpiece} alt="" className="piece"/>
        } else if (this.props.type === "white-queen") {
            image = <img src={whitequeen} alt="" className="piece"/>
        } else if (this.props.type === "black-queen") {
            image = <img src={blackqueen} alt="" className="piece"/>
        } else if (this.props.type === "move") {
            image = <img src={move} alt=""/>
        }

        return image
    }
}

export default Piece;
