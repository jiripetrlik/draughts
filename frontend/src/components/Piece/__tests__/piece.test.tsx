import React from 'react';
import renderer from 'react-test-renderer';
import Piece from '../Piece'

test('snapshot test', () => {
    const type = "white-piece"

    const tree = renderer
        .create(<Piece type={type} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
