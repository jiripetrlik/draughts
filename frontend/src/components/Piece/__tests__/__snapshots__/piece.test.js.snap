// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<img
  alt=""
  src="white-piece.svg"
/>
`;
